### ISSUE TITLE
Tracking issue: Gitserver migration to Rendezvous hashing

### ISSUE THREAD

**Issue by asdine** [sourcegraph/sourcegraph#32311]
### Plan

This is the tracking issue for [RFC 595](https://docs.google.com/document/d/1sZBF6EhFCMu3JTJBIPHP221zhmEDmjZvk_tJdAOS3Eo/edit#).

### Tracked issues

@unassigned: __31.50d__

Completed: __31.50d__
- [x] (🏁 20 days ago) https://github.com/sourcegraph/sourcegraph/issues/32324  __3.00d__
- [x] (🏁 20 days ago) https://github.com/sourcegraph/sourcegraph/issues/32322  __5.00d__
- [x] (🏁 20 days ago) https://github.com/sourcegraph/sourcegraph/issues/32827 (PRs: ~[#32830](https://github.com/sourcegraph/sourcegraph/pull/32830)~, ~[#32831](https://github.com/sourcegraph/sourcegraph/pull/32831)~, ~[#32886](https://github.com/sourcegraph/sourcegraph/pull/32886)~, ~[#32997](https://github.com/sourcegraph/sourcegraph/pull/32997)~, ~[#33003](https://github.com/sourcegraph/sourcegraph/pull/33003)~, ~[#33059](https://github.com/sourcegraph/sourcegraph/pull/33059)~, ~[#33067](https://github.com/sourcegraph/sourcegraph/pull/33067)~, ~[#33071](https://github.com/sourcegraph/sourcegraph/pull/33071)~, [#33072](https://github.com/sourcegraph/sourcegraph/pull/33072), ~[#33210](https://github.com/sourcegraph/sourcegraph/pull/33210)~) __5.00d__
- [x] (🏁 20 days ago) https://github.com/sourcegraph/sourcegraph/issues/32468 
- [x] (🏁 20 days ago) https://github.com/sourcegraph/sourcegraph/issues/32467 
- [x] (🏁 20 days ago) https://github.com/sourcegraph/sourcegraph/issues/32466 
- [x] (🏁 20 days ago) https://github.com/sourcegraph/sourcegraph/issues/32321  __3.00d__
- [x] (🏁 20 days ago) https://github.com/sourcegraph/sourcegraph/issues/32313  __0.50d__
- [x] (🏁 20 days ago) https://github.com/sourcegraph/sourcegraph/issues/32314  __5.00d__
- [x] (🏁 20 days ago) https://github.com/sourcegraph/sourcegraph/issues/32316  __5.00d__
- [x] (🏁 20 days ago) https://github.com/sourcegraph/sourcegraph/issues/32317  __4.00d__
- [x] (🏁 3 days ago) https://github.com/sourcegraph/sourcegraph/issues/32318  __1.00d__

@asdine: __5.00d__

- [ ] https://github.com/sourcegraph/sourcegraph/pull/33072 

Completed: __5.00d__
- [x] (🏁 578 days ago) https://github.com/sourcegraph/sourcegraph/issues/32197 (PRs: ~[#32325](https://github.com/sourcegraph/sourcegraph/pull/32325)~, ~[#32364](https://github.com/sourcegraph/sourcegraph/pull/32364)~, ~[#32369](https://github.com/sourcegraph/sourcegraph/pull/32369)~, ~[#32429](https://github.com/sourcegraph/sourcegraph/pull/32429)~, ~[#32431](https://github.com/sourcegraph/sourcegraph/pull/32431)~) __2.00d__
- [x] (🏁 577 days ago) https://github.com/sourcegraph/sourcegraph/pull/32491 
- [x] (🏁 573 days ago) https://github.com/sourcegraph/sourcegraph/issues/32312 (PRs: ~[#32432](https://github.com/sourcegraph/sourcegraph/pull/32432)~, ~[#32503](https://github.com/sourcegraph/sourcegraph/pull/32503)~) __3.00d__
- [x] (🏁 566 days ago) https://github.com/sourcegraph/sourcegraph/pull/32830 
- [x] (🏁 565 days ago) https://github.com/sourcegraph/sourcegraph/pull/32886 
- [x] (🏁 564 days ago) https://github.com/sourcegraph/sourcegraph/pull/32997 
- [x] (🏁 563 days ago) https://github.com/sourcegraph/sourcegraph/pull/33003 
- [x] (🏁 563 days ago) https://github.com/sourcegraph/sourcegraph/pull/33059 
- [x] (🏁 563 days ago) https://github.com/sourcegraph/sourcegraph/pull/33067 
- [x] (🏁 563 days ago) https://github.com/sourcegraph/sourcegraph/pull/33071 
- [x] (🏁 558 days ago) https://github.com/sourcegraph/sourcegraph/pull/33210 

@sashaostrikov: __3.00d__

Completed: __3.00d__
- [x] (🏁 566 days ago) https://github.com/sourcegraph/sourcegraph/pull/32781 
- [x] (🏁 566 days ago) https://github.com/sourcegraph/sourcegraph/pull/32880 
- [x] (🏁 564 days ago) https://github.com/sourcegraph/sourcegraph/issues/32315 (PRs: ~[#32621](https://github.com/sourcegraph/sourcegraph/pull/32621)~, ~[#32769](https://github.com/sourcegraph/sourcegraph/pull/32769)~, ~[#32890](https://github.com/sourcegraph/sourcegraph/pull/32890)~) __3.00d__
- [x] (🏁 563 days ago) https://github.com/sourcegraph/sourcegraph/pull/33074 
- [x] (🏁 553 days ago) https://github.com/sourcegraph/sourcegraph/issues/33000 (PRs: ~[#33263](https://github.com/sourcegraph/sourcegraph/pull/33263)~)
- [x] (🏁 551 days ago) https://github.com/sourcegraph/sourcegraph/pull/33487 
- [x] (🏁 550 days ago) https://github.com/sourcegraph/sourcegraph/pull/33560 
- [x] (🏁 496 days ago) https://github.com/sourcegraph/sourcegraph/issues/33485 
- [x] (🏁 432 days ago) https://github.com/sourcegraph/sourcegraph/issues/32482 

@vrto: __4.00d__

Completed: __4.00d__
- [x] (🏁 508 days ago) https://github.com/sourcegraph/sourcegraph/issues/32319 (PRs: ~[#32831](https://github.com/sourcegraph/sourcegraph/pull/32831)~, ~[#32831](https://github.com/sourcegraph/sourcegraph/pull/32831)~) __4.00d__

#### Legend

- 👩 Customer issue
- 🐛 Bug
- 🧶 Technical debt
- 🎩 Quality of life
- 🛠️ [Roadmap](https://handbook.sourcegraph.com/departments/product-engineering/process/planning-process#roadmap)
- 🕵️ [Spike](https://en.wikipedia.org/wiki/Spike_(software_development))
- 🔒 Security issue
- 🙆 Stretch goal

**Comment by dan-mckean** [comment]
Suggested weekly summary headings:
- Progress since last update
- Overall progress towards goal/completion
- Any blockers/learnings/changes to the plan since last update

**Comment by asdine** [comment]
### Last week

We started the ground work:
- Storing all repositories size in the database, which will allow us to statically calculate the disk size required for each gitserver instance during the migration. https://github.com/sourcegraph/sourcegraph/issues/32197
- Added an internal Cursor API that will be used by both the migration and the gitserver clients. This is useful to parallelize the work between us. https://github.com/sourcegraph/sourcegraph/pull/32491
- Finished the work needed to allow cloning a repo from one gitserver instance to another. Now this endpoint can be used by the migration to move things around. https://github.com/sourcegraph/sourcegraph/issues/32312

### This week

We will work in parallel:

- @asdine: Write the migration (https://github.com/sourcegraph/sourcegraph/issues/32314) and other satellite tasks
- @sashaostrikov: Work on the gitserver client https://github.com/sourcegraph/sourcegraph/issues/32315

### Progress and blockers

The work has just started last week but everything went smoothly.
The team however is not at 100% capacity.

**Comment by asdine** [comment]
### Last week

- @asdine was on support the whole week
- @sashaostrikov Worked on injecting the database handle everywhere the gitserver is used. https://github.com/sourcegraph/sourcegraph/pull/32769


### This week

- @asdine Write the migration (https://github.com/sourcegraph/sourcegraph/issues/32314)
- @sashaostrikov Work on the gitserver client https://github.com/sourcegraph/sourcegraph/issues/32315

**Comment by asdine** [comment]
### Last week

- @asdine Implemented a repository replicator worker https://github.com/sourcegraph/sourcegraph/issues/32827
- @sashaostrikov Redesign gitserver client to take a database parameter and read the cursor

### This week

- @asdine 
  - Write the migration https://github.com/sourcegraph/sourcegraph/issues/32314
 - @sashaostrikov Ensure the gitserver_repos table contains the repo size for all repos https://github.com/sourcegraph/sourcegraph/issues/33000
